### PR TITLE
Updates for Windows

### PR DESCRIPTION
**Pull request by MingweiSamuel**

### CODE REVIEW

File: hydro_deploy/core/src/hydroflow_crate/build.rs [hunk 1]
```diff
@@ -122,22 +126,52 @@ pub async fn build_crate(
...
-                .map_err(|_| "failed to spawn blocking task")?
+                .map_err(|_| BuildError::TokioJoinError)?
             })
         })
         .await
         .cloned()
 }
+
+#[derive(Clone, Debug)]
+pub enum BuildError {
```
Comment (shadaj) [inline hydro_deploy/core/src/hydroflow_crate/build.rs:151]: Nice

**Review by shadaj — APPROVED** [review]

**Comment by shadaj** [comment]
These test failures are strange, it seems that the `cfg(target_os = ...)` is behaving differently when cargo is invoked by Hydro Deploy. Hydro Deploy shouldn't be specifying any target though, we leave it as the default for a `Localhost` host, so I'm a bit confused why this is happening.

**Comment by MingweiSamuel** [comment]
@shadaj I also can't repro (anymore) on my local windows, lol. Removing the CI for now

**Comment by shadaj** [comment]
I wonder if we need to be using `which` to grab the right `cargo` from the PATH, I vaguely remember needing to do something similar when invoking processes in another project.

**Comment by MingweiSamuel** [comment]
Actually can reproduce again locally. Build seems to be broken in typenum https://github.com/paholg/typenum/pull/195 

Will merge this and continue

**Comment by MingweiSamuel** [comment]
Seems like OUT_DIR is getting mutilated:
`D:\Projects\hydroflow\target\debug\build\typenum-27ab1f3e5e6288a0\out` normally
vs
`\\?\D:\Projects\hydroflow\target\release\build\typenum-123f6570ec25fa24\out` when the error occurs

**Comment by MingweiSamuel** [comment]
https://github.com/paholg/typenum/pull/195#issuecomment-1876119227

```
include!(r"D:\Projects\typenum_test/include.rs"); // works
include!(r"\\?\D:\Projects\typenum_test\include.rs"); // works
include!(r"\\?\D:\Projects\typenum_test/include.rs"); // ERROR
```

**Comment by MingweiSamuel** [comment]
https://github.com/rust-lang/cargo/blob/f9946d15dfdc7128f8156c29ef9981f13a9ad959/src/doc/contrib/src/implementation/filesystem.md?plain=1#L12

**Comment by shadaj** [comment]
Yep, https://stackoverflow.com/questions/50322817/how-do-i-remove-the-prefix-from-a-canonical-windows-path. We canonicalize the src path before using it as the current dir for the command.

**Pull request merged**